File: ABC/B/256/B.py
n = int(input())
a = list(map(int, input().split()))

cnt = 0
for i in range(n):
    pos = 0
    for j in range(i, n):
        pos += a[j]
        if pos >= 4:
            cnt += 1
            break
print(cnt)

File: ABC/B/256/B.rb
n = gets.chomp.to_i
a = gets.chomp.split.map(&:to_i)

cnt = 0
for i in 0...n do
    pos = 0
    for j in i...n do
        pos += a[j]
        if pos >= 4 then
            cnt += 1
            break
        end
    end
end
p cnt 